### PR TITLE
[Bots] Convert Load, Save, SaveNew, and Delete to Repositories.

### PR DESCRIPTION
**Pull request by Kinglykrab**
# Notes
- General code cleanup, as manually adding to these queries doesn't scale very well.

### CODE REVIEW

File: zone/bot_database.cpp [hunk 1]
```diff
@@ -379,86 +380,47 @@ bool BotDatabase::LoadBot(const uint32 bot_id, Bot*& loaded_bot)
...
-		std::stoul(row[14]), // Eye Color 1
-		std::stoul(row[15]), // Eye Color 2
-		std::stoul(row[16]), // Beard
-		std::stoul(row[17]), // Beard Color
-		std::stoul(row[18]), // Drakkin Heritage
-		std::stoul(row[19]), // Drakkin Tattoo
-		std::stoul(row[20]), // Drakkin Details
-		std::stoi(row[21]), // Health
-		std::stoi(row[22]), // Mana
+		l[0].spells_id,
```
Comment (Akkadius) [inline zone/bot_database.cpp:404]: Would really prefer we use the `FindOne` method when we intend on working with one object

It looks like the repository template will need to be updated to use the primary key in the `fmt::format`

```cpp
	static BotData FindOne(
		Database& db,
		int bot_data_id
	)
	{
		auto results = db.QueryDatabase(
			fmt::format(
				"{} WHERE id = {} LIMIT 1",
				BaseSelect(),
				bot_data_id
			)
		);
```

Comment (Akkadius) [reply zone/bot_database.cpp:404]: Then you are accessing just the one object via `l->` versus `l[0]->` which is a bit strange to read

File: zone/bot_database.cpp [hunk 2]
```diff
@@ -480,304 +442,181 @@ bool BotDatabase::LoadBot(const uint32 bot_id, Bot*& loaded_bot)
...
+			bot_inst->GetBotID()
+		)
 	);
-	auto results = database.QueryDatabase(query);
-	if (!results.Success())
+	if (l.empty()) {
 		return false;
+	}
+
+	l[0].owner_id               = bot_inst->GetBotOwnerCharacterID();
```
Comment (Akkadius) [inline zone/bot_database.cpp:554]: Same here, similar to feedback prior

**Review by Akkadius — APPROVED** [review]
Minor comments, love to see the repository usage

File: zone/bot_database.cpp (hunk 1 above)
Comment (Kinglykrab) [reply zone/bot_database.cpp:404]: Ah yeah, forgot about FindOne while I was working on this. No worries.

**Pull request merged**